src/client/mod.rs

```diff
@@ -216,12 +254,29 @@ impl Client {
         })
     }
 
-    pub fn begin(&mut self, n_metrics: u64) -> io::Result<&mut Client> {
+    pub fn begin(&mut self, n_indoms: u64, n_instances: u64, n_metrics: u64) -> io::Result<&mut Client> {
```

Comment:
To be honest, this is too weird, and very easy to get wrong, lets follow the builder pattern for InstanceMetric, with add_instance adding an instance